### PR TITLE
Add new rule `require-meta-docs-description`

### PR DESCRIPTION
**Pull request by bmish**
The eslint core plugin has an internal lint rule to enforce this same thing: https://github.com/eslint/eslint/blob/master/tools/internal-rules/consistent-docs-description.js

### CODE REVIEW

**Comment by bmish** [comment]
Hey @not-an-aardvark, I've got several PRs open in this repository, for improvements I'd like to make use of in eslint-plugin-ember which I maintain, and which I'd also like to use in the core eslint plugin to replace some of the custom linting they have.

File: docs/rules/require-meta-docs-description.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# require rules to implement a meta.docs.description property (require-meta-docs-description)
+
+Defining a clear and consistent description for each rule helps developers understand what they're used for.
+
+In particular, each rule description should begin with an allowed prefix:
+* `enforce`
+* `require`
+* `disallow`
```
Comment (not-an-aardvark) [inline docs/rules/require-meta-docs-description.md]: I think this one should probably be configurable with something like a regex (maybe with `enforce`/`require`/`disallow` as the default). As far as I know, `enforce`/`require`/`disallow` is a convention used by the ESLint team but it's not universally adopted.

Comment (bmish) [reply docs/rules/require-meta-docs-description.md]: Good point, I'll update shortly.

Comment (bmish) [reply docs/rules/require-meta-docs-description.md]: I added an `allowedPrefixes` array option to the rule.

**Comment by bmish** [comment]
@not-an-aardvark it seems like an unrelated commit beafe713387f7ef2a6ff844400e231bbbe73c5fb was accidentally added to this PR? That commit was already merged in a separate PR.

**Comment by not-an-aardvark** [comment]
Sorry, that's probably my fault (I force-pushed to master to update a commit message.)

**Comment by bmish** [comment]
Okay I just rebased to fix it. 

Let me know what you think of the PR and then I'd be interested to get a new release once this is merged.

**Comment by bmish** [comment]
@not-an-aardvark ping?

File: docs/rules/require-meta-docs-description.md (hunk 1 above)
Comment (not-an-aardvark) [reply docs/rules/require-meta-docs-description.md]: I'd really rather this be a regex option -- choosing a set of specific prefixes feels like an oddly specific restriction to impose on users of the rule.

Comment (bmish) [reply docs/rules/require-meta-docs-description.md]: @not-an-aardvark sounds good, I switched to a `pattern` option.

**Comment by bmish** [comment]
Thanks for the review @not-an-aardvark! Can we get a release?

**Comment by not-an-aardvark** [comment]
Published as v2.2.0, thanks for your patience

**Pull request merged**